### PR TITLE
Optimize brain::Circuit::loadMorphologies

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: brain/circuit.cpp [hunk 1]
```diff
@@ -98,19 +101,15 @@ neuron::Morphologies Circuit::loadMorphologies( const GIDSet& gids,
     GIDSet::const_iterator gid = gids.begin();
     for( size_t i = 0; i < uris.size(); ++i, ++gid )
```
Comment (favreau) [inline brain/circuit.cpp]: Can you use C++11 syntax there? for( const auto& gid: gids)

Comment (hernando) [inline brain/circuit.cpp]: No because there are counters being incremented in the loop

Comment (favreau) [inline brain/circuit.cpp]: I know but simply move ++gid inside the loop, and that gets rid of the GIDSet::const_iterator. Anyway, just a suggestion there.

Comment (hernando) [inline brain/circuit.cpp]: Up to @eile to decide, I tend to avoid increments inside the loop body. On the other hand: auto gid = gids.begin(); is welcome.

**Pull request merged**